oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -336,6 +339,37 @@ private boolean indexProperty(String path,
         return dirty;
     }
 
+    static String getValueAsTruncatedString(Object val) {
+        String value = val.toString();
+        boolean isArray = value.startsWith("[") && value.endsWith("]");
+        return value.length() <= LOG_TRUNCATION_LENGTH ? value
```

Comment:
Personal opinion: in this situation, I find the ternary operator harder to read than an if-then-else. It was also the reason why I misunderstood the code initially, and thought that there was a risk of out of bounds exception. But it's a matter of personal preference, so feel free to leave it as it is.

Reply:
I've modified the function now - should be more clear.